### PR TITLE
add update/put class endpoint

### PR DESCRIPTION
**Pull request by brandonwong368**
### Notion ticket link

[PUT endpoint for Class](https://www.notion.so/uwblueprintexecs/086c1b564dbf478faa649a8880e34b81?v=9d3f8002a0d1477ba395616dfc082af7&p=a98c5dc6487e4bb9a9a4c9c771d9e04f)

### Implementation description

- created function in database services to update a class
- body is passed in and validated, if valid class is updated, if invalid, error is thrown

### Steps to test

1. Postman was used to test the class update

image

### What should reviewers focus on?

-

### Checklist

-   [x] My PR name is descriptive and in imperative tense
-   [x] I have run the linter
-   [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/api/class/[id].ts [hunk 1]
```diff
@@ -33,8 +35,29 @@ export default async function handle(
         }
         ResponseUtil.returnOK(res, deletedClass);
         return;
+    } else if (req.method == "PUT") {
+        // validate updated class body
+        const validationError = validateClassData(req.body as ClassInput);
```
Comment (feng-eric) [inline pages/api/class/[id].ts]:
```suggestion
        const classInput = req.body as ClassInput;
        const validationError = validateClassData(classInput);
```

File: pages/api/class/[id].ts [hunk 2]
```diff
@@ -33,8 +35,29 @@ export default async function handle(
...
+        if (validationError.length !== 0) {
+            ResponseUtil.returnBadRequest(res, validationError.join(", "));
+            return;
+        }
+        // obtain class id
+        const { id } = req.query;
+        // obtain the updated class body
+        const updatedClass = await updateClass(
+            id as string,
+            req.body as ClassInput,
```
Comment (feng-eric) [inline pages/api/class/[id].ts]:
```suggestion
            classInput,
```

File: models/Class.ts [hunk 3]
```diff
@@ -3,7 +3,8 @@ import { weekday } from "@prisma/client";
 /**
  * Request Body Input for POST /class
  */
-export type CreateClassInput = {
+export type ClassInput = {
+    // should id be upd
```
Comment (feng-eric) [inline models/Class.ts]: What's. this comment?

File: prisma/schema.prisma [hunk 4]
```diff
@@ -132,8 +132,8 @@ model TeacherReg {
 
 model User {
   id            Int           @id @default(autoincrement())
-  firstName     String?       @map("first_name")
```
Comment (feng-eric) [inline prisma/schema.prisma]: Why did you change this?

File: services/database/class.ts [hunk 5]
```diff
@@ -62,4 +62,22 @@ async function deleteClass(id: string): Promise<Class> {
...
 }
 
-export { getClass, getClasses, createClass, deleteClass };
+/**
+ * updateClass takes in id of the class and updates the class information
+ * @param id - classId of the class to be updated
+ * @returns Promise<Class> - Promise with the updated class
+ */
+async function updateClass(
+    id: string,
```
Comment (feng-eric) [inline services/database/class.ts]: @mahadzaryab1 Should we convert `id` to a number prior?

**Review by feng-eric — COMMENTED** [review]
Left some comments

File: prisma/schema.prisma (hunk 4 above)
Comment (brandonwong368) [reply prisma/schema.prisma]: I'm not sure, when I ran commands I think it auto updated, should it just be the same as the schema.prisma file in staging?

Comment (feng-eric) [reply prisma/schema.prisma]: It should be `firstName` with the mapping. Is your schema.sql file up to date?

File: services/database/class.ts (hunk 5 above)
Comment (feng-eric) [reply services/database/class.ts]: @brandonwong368 Can you leave a comment so Mahad can see it later?

File: pages/api/class/[id].ts [hunk 6]
```diff
@@ -33,8 +35,27 @@ export default async function handle(
...
+    } else if (req.method == "PUT") {
+        // validate updated class body
+        const classInput = req.body as ClassInput;
+        const validationError = validateClassData(classInput);
+        if (validationError.length !== 0) {
+            ResponseUtil.returnBadRequest(res, validationError.join(", "));
+            return;
+        }
+        // obtain class id
+        const { id } = req.query;
```
Comment (feng-eric) [inline pages/api/class/[id].ts]: I think we should perhaps validate it prior to passing it into `updateClass`. See below for reference:

https://github.com/uwblueprint/social-diversity-for-children/blob/9a6325a317bda14b4033bb75ac94c53592db5c6c/pages/api/enroll/child.ts#L34-L44

Comment (feng-eric) [reply pages/api/class/[id].ts]: Also, isn't `id` already being retrieved from req.query on line 18?

Comment (feng-eric) [reply pages/api/class/[id].ts]: We can do this validation right after line 18 as for both GET and PUT, we want to make sure `id` is a number.

Comment (feng-eric) [reply pages/api/class/[id].ts]: After you make that change, you can update function signature for `updateClass` to accept id as a number instead!

File: services/database/class.ts (hunk 5 above)
Comment (brandonwong368) [reply services/database/class.ts]: @mahadzaryab1 bump

**Review by feng-eric — APPROVED** [review]
